File: platform/WinForms/SweetEditor/CoreLinkedEditing.cs
#nullable enable
using System;
using System.Collections.Generic;

namespace SweetEditor {

    public sealed partial class LinkedEditingModel {
        public List<TabStopGroup> Groups { get; set; } = new();
    }

    public sealed partial class TabStopGroup {
        public int Index { get; set; } = 0;
        public List<TextRange> Ranges { get; set; } = new();
        public string DefaultText { get; set; } = string.Empty;
    }
}

File: platform/Apple/SweetEditor-Shared/core/CoreLinkedEditing.swift
import Foundation

public struct LinkedEditingModel {
    public var groups: [TabStopGroup] = []

    public init(groups: [TabStopGroup] = []) {
        self.groups = groups
    }
}

public struct TabStopGroup {
    public var index: Int32 = 0
    public var ranges: [TextRange] = []
    public var default_text: String = ""

    public init(index: Int32 = 0, ranges: [TextRange] = [], default_text: String = "") {
        self.index = index
        self.ranges = ranges
        self.default_text = default_text
    }
}
